srtcore/core.cpp

```diff
@@ -8034,7 +8093,10 @@ std::pair<int, steady_clock::time_point> CUDT::packData(CPacket& w_packet)
     }
 
     w_packet.m_iID = m_PeerID;
-    w_packet.setLength(payload);
+
+   // Redundant. This is now properly set in both rexmit and normal
+   // cases just after setting the m_pcData field.
+   //w_packet.setLength(payload);
```

Comment:
This is because the reference to a `CPacket` object is passed to `CSndBuffer::readData` in both cases.

Reply:
Just delete this code then.